### DISCUSSION TITLE
Semantic versioning including patch for all releases?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [checkstyle/checkstyle#12436] [Q&A] [answered]
Would you mind creating all releases with a patch number? E.g. 10.4.0 instead of 10.4? I'm not sure if the checkstyle project follows SemVer (which does require 3 part version numbers). The practical reason for asking is that it's just harder to upgrade dependencies if you have to remember which ones are 3 part versions and which are 2 part versions (and the majority of stuff in maven is 3 parts also for x.y releases). Trust me, I have experienced it multiple times that I changed a version number for checkstyle in my own poms from something like X.Y.Z to X.Y+1.0, just to see the build fail, and then to remember it doesn't have the zero there. You would make my life (and that of many others) a tiny bit more simple by always going with 3 version parts. :)

**Comment by rnveach** [comment]
https://checkstyle.org/#Development_Cycle
> Checkstyle is not using Semantic Versioning ([reasons in issue #3709](https://github.com/checkstyle/checkstyle/issues/3709))

> 10.4.0 instead of 10.4

There is a discussion on this somewhere, I don't know if there has been an outcome yet.

**Reply by romani** [reply]
I do not know why we use 2 digits, but it was even before me.
It use to be convenient for us, as 90% of our releases (based on our versioning) as 2nd digit bump. so 10.1 auto updates to 10.2 by maven.
but right now we actively changing this, thanks a lot for help of @stoyanK7 to improve release automation of our project.

usage of 3 digits become required for us.
10.1.0 will be updated to 10.1.1 by maven and it will be what we need by our release version validation.

some discussion on this is at: https://github.com/checkstyle/checkstyle/pull/12316#issuecomment-1303756956

All we need to make new habbit is to use 3 digits all the time.
...... and we need to change out release notes validation as it is based on amount of dots :) .
@stoyanK7 , please help us to improve - https://github.com/checkstyle/contribution/issues/701

**Comment by romani** [comment, marked as answer]
version is changed to 3 digits, will always be 3 digits.
10.5.0 - will be released on 27 Nov 2022

**Reply by romani** [reply]
https://checkstyle.org/releasenotes.html#Release_10.5.0

@Bananeweizen , please resolve question.

**Reply by Bananeweizen** [reply]
That's not possible. I didn't know it before you requested that, but it actually depends on the discussion category, whether that action is available. It's not available here in "ideas". https://github.com/community/community/discussions/2903

**Reply by nrmancuso** [reply]
Moved to Q&A, and selected answer.
